src/main/java/com/aol/cyclops2/data/collections/extensions/IndexedSequenceX.java

```diff
@@ -101,8 +104,21 @@
     @Override
     <R> IndexedSequenceX<R> flatMap(Function<? super T, ? extends Iterable<? extends R>> mapper);
 
-    
-    
-    
-    
+    /**
+     * Intercalate
+     *
+     * @param listOfLists Index of our sub sequence (inclusive)
+     * @return List with current IndexedSequenceX inserted between each List.
+     */
+    default List intercalate(List<List> listOfLists) {
+       List intercalated = null;
+        for (List list : listOfLists) {
```

Comment:
Another challenge would be to make this execute lazily. I think you may be able to do it via intersperse (intersperse this IndexedSequenceX and then flatMap to flatten the List). If you convert listOfLists to a ListX (ListX.fromIterable(listOfLists) ) you will be able to access the intersperse operator there..